compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -683,16 +683,16 @@ trait Implicits { self: Typer =>
         case arg :: Nil if !arg.typeSymbol.is(Param) =>
           object bindFreeVars extends TypeMap {
             var ok = true
-            def apply(t: Type) = t match {
-              case t @ TypeRef(NoPrefix, _) =>
+            def apply(t: Type) = t.dealias match {
+              case t @ TypeRef(NoPrefix, _) if !ctx.inInlineMethod =>
                 inferImplicit(defn.QuotedTypeType.appliedTo(t), EmptyTree, span) match {
                   case SearchSuccess(tag, _, _) if tag.tpe.isStable =>
                     tag.tpe.select(defn.QuotedType_splice)
                   case _ =>
                     ok = false
                     t
                 }
-              case _ => t
+              case tp => mapOver(tp)
```

Comment:
what is the motivation to `mapOver`? This is suspicious. Is it to dealias further term as you have in the tests? Mapping over may also trigger another implicit search for parts of the type which should only be done in the PCP checks.

Reply:
If we do not `mapOver`, in staging, for the following code:

```Scala
import scala.quoted._

inline def foo[T:Type]: Int = 10
def main = {
  type S = Int
  type V = List[S]
  foo[V]
}
```

It will generate an evidence in `staging`:

```Scala
        type V = List[S]
        {
          val evidence$1: quoted.Type[List[S]] =
            '[List[quoted.Type[Int]#$splice]]
          10:Int
        }
```

The code will then crash `ReifyQuotes.scala:153`, as it assumes all evidences are `TermRef`s.